tests/plugins/dialogadvtab/dialog.js

```diff
@@ -68,7 +76,64 @@
 
 				dialog.hide();
 			} );
+		},
+
+		// (#1046).
+		'test adding two subsequent id changes': function() {
+			var bot = this.editorBot;
+
+			this.editorBot.setHtmlWithSelection( '<p>[aaa] <strong>bbb</strong></p>' );
+
+			// First add id to one of links.
+			bot.dialog( 'link', function( dialog ) {
+				var editor = this.editor;
+
+				dialog.setValueOf( 'info', 'url', 'a' );
+				dialog.setValueOf( 'advanced', 'advId', 'aa' );
+				dialog.getButton( 'ok' ).click();
+
+				assert.beautified.html( '<p><a href="http://a" id="aa">aaa</a> <strong>bbb</strong></p>',
+					bot.getData( true ) );
+
+				editor.getSelection().selectElement( editor.editable().findOne( 'strong' ) );
+
+				// Now add the id to a second link.
+				bot.dialog( 'link', function( dialog ) {
+					dialog.setValueOf( 'info', 'url', 'b' );
+					dialog.setValueOf( 'advanced', 'advId', 'bb' );
+					dialog.getButton( 'ok' ).click();
+
+					assert.beautified.html( '<p><a href="http://a" id="aa">aaa</a> <a href="http://b" id="bb"><strong>bbb</strong></a></p>',
+						bot.getData( true ) );
+				} );
+			} );
+		},
+
+		'test editing two subsequent id changes': function() {
```

Comment:
This TC is not related to this issue, as this particular case was not buggy at all. I have added it only because I saw there was no coverage for such case, and this issue shows that it also should be covered. 🙂